### PR TITLE
Adding test for metro async combination

### PR DESCRIPTION
**Pull request by gejain**
Adding test for metro async combination

**What this PR does / why we need it**:

Adding test for metro async combination -

* Added support for more than 2 kubeconfigs
* Added support for bidirectional clusterpairing in torpedo
* Create bidirectional clusterpair from 1st to 2nd cluster for metro
* Do MetroDR from 1st to 2nd
* Create bidirectional clusterpair from 2nd to 3rd cluster for async

**Which issue(s) this PR fixes** (optional)
Closes #https://purestorage.atlassian.net/browse/PTX-23443

**Special notes for your reviewer**:
Ran locally, need to create a job with 3 clusters for this test, Its in progress.
Local run -
INFO[2024-04-17 11:58:20] Verifying : Description : Test completed successfully ?
INFO[2024-04-17 11:58:20] Actual:PASS, Expected: PASS
INFO[2024-04-17 11:58:21] --------Test End------
INFO[2024-04-17 11:58:21] #Test: MigrateDeploymentMetroAsync
INFO[2024-04-17 11:58:21] #Description: Migration of application using metro+async combination
INFO[2024-04-17 11:58:21] ------------------------
INFO[2024-04-17 11:58:21] Dashboard URL : https://aetos.pwx.purestorage.com/resultSet/testSetID/616051

Ran 1 of 330 Specs in 453.075 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 329 Skipped

```

root•~/tornew/torpedo(topic/geet/metroasyncnewway⚡)» s1 get migration -n nginx-metro-async-dr-mig-0-04-17-11h50m47s
NAME                  CLUSTERPAIR         STAGE   STATUS       VOLUMES   RESOURCES   CREATED               ELAPSED                                TOTAL BYTES TRANSFERRED
metro-dr-011h11m55s   remoteclusterpair   Final   Successful   0/0       7/7         17 Apr 24 11:52 UTC   Volumes () Resources (18.583921698s)   0
root•~/tornew/torpedo(topic/geet/metroasyncnewway⚡)» s2 get migration -n nginx-metro-async-dr-mig-0-04-17-11h50m47s         [12:58:59]
NAME                  CLUSTERPAIR            STAGE   STATUS       VOLUMES   RESOURCES   CREATED               ELAPSED                         TOTAL BYTES TRANSFERRED
async-dr-011h11m50s   remoteclusterpairnew   Final   Successful   2/2       7/7         17 Apr 24 11:56 UTC   Volumes (1m8s) Resources (3s)   69070848
root•~/tornew/torpedo(topic/geet/metroasyncnewway⚡)» s2 get clusterpair -n nginx-metro-async-dr-mig-0-04-17-11h50m47s
NAME                   STORAGE-STATUS   SCHEDULER-STATUS   CREATED
remoteclusterpair      NotProvided      Ready              17 Apr 24 11:52 UTC
remoteclusterpairnew   Ready            Ready              17 Apr 24 11:56 UTC
root•~/tornew/torpedo(topic/geet/metroasyncnewway⚡)» s1 get clusterpair -n nginx-metro-async-dr-mig-0-04-17-11h50m47s
NAME                STORAGE-STATUS   SCHEDULER-STATUS   CREATED
remoteclusterpair   NotProvided      Ready              17 Apr 24 11:52 UTC
root•~/tornew/torpedo(topic/geet/metroasyncnewway⚡)»       
``` 
Job -
https://jenkins.pwx.dev.purestorage.com/job/Stork/job/stork-metro-async-comb/25/

### CODE REVIEW

**Review by Rohit-PX — APPROVED** [review]
LGTM.

File: tests/basic/async_dr_test.go [hunk 1]
```diff
@@ -149,6 +153,135 @@ var _ = Describe("{MigrateDeployment}", func() {
 	})
 })
 
+var _ = Describe("{MigrateDeploymentMetroAsync}", func() {
+	var testrailID = 297595
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/297595
+	var runID int
+	var kubeConfigWritten bool
```
Comment (kphalgun-px) [inline tests/basic/async_dr_test.go:160]: Should this be a global variable that is accessible to all test cases?

File: tests/basic/async_dr_test.go [hunk 2]
```diff
@@ -149,6 +153,135 @@ var _ = Describe("{MigrateDeployment}", func() {
 	})
 })
 
+var _ = Describe("{MigrateDeploymentMetroAsync}", func() {
```
Comment (kphalgun-px) [inline tests/basic/async_dr_test.go:156]: Can we use ValidateApplications after switching to the second cluster to see if the apps are up and running?

File: tests/common.go [hunk 3]
```diff
@@ -23,15 +23,38 @@ import (
 	"strings"
 	"sync"
 	"time"
+	context1 "context"
+	// import ssh driver to invoke it's init
```
Comment (kphalgun-px) [inline tests/common.go:27]: I think the comments are misformatted.

File: tests/common.go [hunk 4]
```diff
@@ -3779,6 +3771,148 @@ func CreateClusterPairFile(pairInfo map[string]string, skipStorage, resetConfig
 	return addStorageOptions(pairInfo, clusterPairFileName)
 }
 
+func ScheduleBidirectionalClusterPair(cpName, cpNamespace, projectMappings string, objectStoreType storkv1.BackupLocationType, secretName string, mode string, sourceCluster int, destCluster int) error {
```
Comment (kphalgun-px) [inline tests/common.go:3774]: How about (err error)?

```go
	defer func() {
	        var config *rest.Config // I am not sure
		config, err = clientcmd.BuildConfigFromFlags("", srcKubeConfigPath)
		if err != nil {
			return
		}
		core.Instance().SetConfig(config)
		apps.Instance().SetConfig(config)
		stork.Instance().SetConfig(config)
	}()
```

File: tests/common.go [hunk 5]
```diff
@@ -3779,6 +3771,148 @@ func CreateClusterPairFile(pairInfo map[string]string, skipStorage, resetConfig
...
+			Labels: map[string]string{
+				"creator": "stork-test",
+			},
+		},
+	})
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return fmt.Errorf("Failed to create namespace %s on destination cluster", cpNamespace)
+	}
+
+	destKubeConfigPath, err := GetCustomClusterConfigPath(destCluster)
```
Comment (kphalgun-px) [inline tests/common.go:3826]: return err?

File: tests/common.go [hunk 6]
```diff
@@ -3779,6 +3771,148 @@ func CreateClusterPairFile(pairInfo map[string]string, skipStorage, resetConfig
...
+		    "--kubeconfig", srcKubeConfigPath,
+			"--src-kube-file", srcKubeConfigPath,
+			"--dest-kube-file", destKubeConfigPath,
+			"--mode", "sync-dr",
+	    }
+	}
+
+	if projectMappings != "" {
+		cmdArgs = append(cmdArgs, "--project-mappings")
+		cmdArgs = append(cmdArgs, projectMappings)
```
Comment (kphalgun-px) [inline tests/common.go:3853]: For brevity

```go
cmdArgs = append(cmdArgs, "--project-mappings", projectMappings)
```

Comment (gejain) [reply tests/common.go:3853]: We tried this and it fails sometimes with some space left out, so above one is the existing way which I used.

File: tests/basic/async_dr_test.go (hunk 2 above)
Comment (gejain) [reply tests/basic/async_dr_test.go:156]: We will implement that validation method soon in upcoming phase, as we have to add deactivation on source and activation on destination.

File: tests/basic/async_dr_test.go (hunk 1 above)
Comment (gejain) [reply tests/basic/async_dr_test.go]: Done, Thanks!

File: tests/common.go (hunk 3 above)
Comment (gejain) [reply tests/common.go]: Done, Thanks!

File: tests/common.go (hunk 4 above)
Comment (gejain) [reply tests/common.go]: Done, Thanks!

File: tests/common.go (hunk 5 above)
Comment (gejain) [reply tests/common.go:3850]: Done, Thanks!

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Pull request merged**